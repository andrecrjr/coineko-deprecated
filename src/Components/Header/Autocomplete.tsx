import { Link } from 'react-router-dom';
import { CurrencyItem, searchType } from 'src/Types';
import { useFetch } from 'src/state/Hooks/useSWR';
import { StarPortfolioCurrency } from '../Table/Star';

export const AutoComplete = ({ searchParam }: { searchParam: string }) => {
	const { data, isLoading } = useFetch<searchType>(
		`search?query=${searchParam}`,
		'get'
	);
	if (!isLoading)
		return (
			<ul className="suggest--box" role="listbox" aria-autocomplete="list">
				{data &&
					data.coins.map((item: CurrencyItem) => {
						return (
							<li role="option" key={item.id} className="suggest--box__option">
								<AutoCompleteItem currency={item} />
							</li>
						);
					})}
			</ul>
		);
	return <></>;
};

const AutoCompleteItem = ({ currency }: { currency: CurrencyItem }) => {
	return (
		<div className="grid grid-cols-[25px_auto_auto] my-8 sm:my-3 mx-5 items-center">
			<StarPortfolioCurrency currencyId={currency.id} />
			<img src={currency.thumb} className="ml-4" width="25" height="25" />
			<h3 className="text-sm break-before-auto overflow-hidden ml-4">
				{currency.name}
			</h3>
		</div>
	);
};
