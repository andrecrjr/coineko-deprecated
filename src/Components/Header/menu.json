[
	{ "alias": "Cryptocurrency" },
	{ "alias": "NFT" },
	{
		"alias": "DeFi"
	},
	{ "alias": "Exchange" },
	{
		"alias": "Portfolio"
	}
]