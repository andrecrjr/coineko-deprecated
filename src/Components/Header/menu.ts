export const MenuOptions = [
	{ alias: "Cryptocurrency" },
	{ alias: "NFT" },
	{
		alias: "DeFi",
	},
	{ alias: "Exchange" },
	{
		alias: "Portfolio",
	},
];
