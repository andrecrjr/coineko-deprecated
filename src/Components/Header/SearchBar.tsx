import { FormEvent, useEffect, useState } from 'react';
import { AutoComplete } from './Autocomplete';
import { MdSearch } from 'react-icons/md';

export const SearchBar = () => {
	const [search, setSearch] = useState('');
	const [suggestionItens, setSuggest] = useState<string>('');
	useEffect(() => {
		if (suggestionItens.length > 0) {
			setTimeout(() => {
				setSuggest('');
			}, 10000);
		}
	}, [suggestionItens]);

	const updateSuggestion = (
		e: FormEvent<HTMLInputElement | HTMLFormElement>
	) => {
		e.preventDefault();
		setSuggest(search);
	};

	return (
		<form className="relative w-full" onSubmit={updateSuggestion}>
			<input
				type="text"
				className="search--box"
				placeholder="Search Currency or Exchange"
				value={search}
				aria-label="search"
				role="combobox"
				onChange={(e) => setSearch(e.target.value)}
				onBlur={updateSuggestion}
			/>
			<button type="submit" className="absolute top-0 right-0">
				<MdSearch width="30" height="30" />
			</button>
			{suggestionItens !== '' && <AutoComplete searchParam={suggestionItens} />}
		</form>
	);
};
