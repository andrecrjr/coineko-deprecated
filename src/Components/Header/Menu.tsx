import React, { useContext } from 'react';
import { MenuOptions } from './menu';
import { PaginationContext } from 'src/state/Contexts';
import { Link, useLocation } from 'react-router-dom';

const Menu = () => {
	const { setPagination } = useContext(PaginationContext);
	const { pathname } = useLocation();

	return (
		<nav
			className="w-full bg-purple-neko overflow-x-scroll 
			h-8 sm:h-10 flex content-center sm:justify-center sm:overflow-auto"
		>
			<ul className="list-none flex self-center ">
				{MenuOptions.map((item, index) => (
					<Link
						to={{
							pathname: item.path
						}}
						key={index}
						onClick={() => {
							setPagination((oldState) => ({
								...oldState,
								...{
									number: 1,
									firstOrPagination: 'current'
								}
							}));
						}}
					>
						<li
							className={`text-sm pr-10 leading-5 first:pl-4 last:pr-4 font-roboto 
                        ${
													pathname.replace('/', '') === item.path &&
													'font-bold underline'
												} text-white `}
							data-testid={`button-${item.alias.toLowerCase()}`}
						>
							{item.alias}
						</li>
					</Link>
				))}
			</ul>
		</nav>
	);
};

export default Menu;
