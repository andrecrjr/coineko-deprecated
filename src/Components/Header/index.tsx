import { MenuOptions } from "./menu";
import Cat from "../../assets/cat.svg?component";

export const Header = () => {
  return (
    <header className="flex flex-col items-center">
      <section className="ml-4 py-2 flex w-10/12">
        <h1 className="text-4xl flex">
          coinek
          <span>
            <Cat className="w-10" />
          </span>
        </h1>
      </section>
      <nav
        className="w-full bg-purple-neko overflow-x-scroll 
				h-8 sm:h-10 flex content-center sm:justify-center sm:overflow-auto"
      >
        <ul className="list-none flex self-center ">
          {MenuOptions.map((item) => (
            <li className="text-sm pr-10 leading-5 first:pl-4 last:pr-4 font-roboto">
              {item.alias}
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
};
