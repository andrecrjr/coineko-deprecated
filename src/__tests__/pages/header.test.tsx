import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { SearchBar } from 'src/Components/Header/SearchBar';
import { describe, expect, it, vi } from 'vitest';
import searchMock from 'src/__mocks__/searchCrypto.mock.json';

vi.mock('axios', async () => {
	const create = vi.fn().mockImplementation(({ baseURL }) => {
		return {
			get: vi.fn().mockImplementation((url) => {
				const apiFilterUrl: string = baseURL + url;
				if (
					apiFilterUrl.includes(
						'https://api.coingecko.com/api/v3/search?query=Bit'
					)
				) {
					return { data: searchMock };
				}
			})
		};
	});

	return { default: { create } };
});

describe('Header test', () => {
	it('should input searchbox', () => {
		render(<SearchBar />);
		const input = screen.getByPlaceholderText(
			'Search Currency or Exchange'
		) as HTMLInputElement;
		fireEvent.change(input, { target: { value: 'Bit' } });
		expect(input.value).toBe('Bit');
	});

	it('should render with auto complete in searchbox', async () => {
		const { container } = render(<SearchBar />);
		const input = screen.getByPlaceholderText(
			'Search Currency or Exchange'
		) as HTMLInputElement;
		fireEvent.change(input, { target: { value: 'Bit' } });
		fireEvent.blur(input);
		const suggestBox = await screen.findAllByRole('option');
		expect(suggestBox[0].textContent).toBe('Bitcoin');
		expect(suggestBox[1].textContent).not.toBe('Ethereum');
		expect(suggestBox[2].textContent).toBe('Bitcoin Cash');
		expect(container.children).toMatchSnapshot();
	});
});
