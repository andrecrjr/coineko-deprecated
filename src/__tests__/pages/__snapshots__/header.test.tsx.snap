// Vitest Snapshot v1

exports[`Header test > should render with auto complete in searchbox 1`] = `
HTMLCollection [
  <form
    class="relative w-full"
  >
    <input
      aria-label="search"
      class="search--box"
      placeholder="Search Currency or Exchange"
      role="combobox"
      type="text"
      value="Bit"
    />
    <button
      class="absolute top-0 right-0"
      type="submit"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </button>
    <ul
      aria-autocomplete="list"
      class="suggest--box"
      role="listbox"
    >
      <li
        class="suggest--box__option"
        role="option"
      >
        <div
          class="grid grid-cols-[25px_auto_auto] my-8 sm:my-3 mx-5 items-center"
        >
          <svg
            class="w-6 fill-[none]"
            data-crypto="bitcoin"
            data-testid="favorite-crypto"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.049 2.927c.3-.921 1.603-.921 1.902 0l1.519 4.674a1 1 0 0 0 .95.69h4.915c.969 0 1.371 1.24.588 1.81l-3.976 2.888a1 1 0 0 0-.363 1.118l1.518 4.674c.3.922-.755 1.688-1.538 1.118l-3.976-2.888a1 1 0 0 0-1.176 0l-3.976 2.888c-.783.57-1.838-.197-1.538-1.118l1.518-4.674a1 1 0 0 0-.363-1.118l-3.976-2.888c-.784-.57-.38-1.81.588-1.81h4.914a1 1 0 0 0 .951-.69l1.519-4.674z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <img
            class="ml-4"
            height="25"
            src="https://assets.coingecko.com/coins/images/1/thumb/bitcoin.png"
            width="25"
          />
          <h3
            class="text-sm break-before-auto overflow-hidden ml-4"
          >
            Bitcoin
          </h3>
        </div>
      </li>
      <li
        class="suggest--box__option"
        role="option"
      >
        <div
          class="grid grid-cols-[25px_auto_auto] my-8 sm:my-3 mx-5 items-center"
        >
          <svg
            class="w-6 fill-[none]"
            data-crypto="wrapped-bitcoin"
            data-testid="favorite-crypto"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.049 2.927c.3-.921 1.603-.921 1.902 0l1.519 4.674a1 1 0 0 0 .95.69h4.915c.969 0 1.371 1.24.588 1.81l-3.976 2.888a1 1 0 0 0-.363 1.118l1.518 4.674c.3.922-.755 1.688-1.538 1.118l-3.976-2.888a1 1 0 0 0-1.176 0l-3.976 2.888c-.783.57-1.838-.197-1.538-1.118l1.518-4.674a1 1 0 0 0-.363-1.118l-3.976-2.888c-.784-.57-.38-1.81.588-1.81h4.914a1 1 0 0 0 .951-.69l1.519-4.674z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <img
            class="ml-4"
            height="25"
            src="https://assets.coingecko.com/coins/images/7598/thumb/wrapped_bitcoin_wbtc.png"
            width="25"
          />
          <h3
            class="text-sm break-before-auto overflow-hidden ml-4"
          >
            Wrapped Bitcoin
          </h3>
        </div>
      </li>
      <li
        class="suggest--box__option"
        role="option"
      >
        <div
          class="grid grid-cols-[25px_auto_auto] my-8 sm:my-3 mx-5 items-center"
        >
          <svg
            class="w-6 fill-[none]"
            data-crypto="bitcoin-cash"
            data-testid="favorite-crypto"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.049 2.927c.3-.921 1.603-.921 1.902 0l1.519 4.674a1 1 0 0 0 .95.69h4.915c.969 0 1.371 1.24.588 1.81l-3.976 2.888a1 1 0 0 0-.363 1.118l1.518 4.674c.3.922-.755 1.688-1.538 1.118l-3.976-2.888a1 1 0 0 0-1.176 0l-3.976 2.888c-.783.57-1.838-.197-1.538-1.118l1.518-4.674a1 1 0 0 0-.363-1.118l-3.976-2.888c-.784-.57-.38-1.81.588-1.81h4.914a1 1 0 0 0 .951-.69l1.519-4.674z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <img
            class="ml-4"
            height="25"
            src="https://assets.coingecko.com/coins/images/780/thumb/bitcoin-cash-circle.png"
            width="25"
          />
          <h3
            class="text-sm break-before-auto overflow-hidden ml-4"
          >
            Bitcoin Cash
          </h3>
        </div>
      </li>
      <li
        class="suggest--box__option"
        role="option"
      >
        <div
          class="grid grid-cols-[25px_auto_auto] my-8 sm:my-3 mx-5 items-center"
        >
          <svg
            class="w-6 fill-[none]"
            data-crypto="bitcoin-cash-sv"
            data-testid="favorite-crypto"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.049 2.927c.3-.921 1.603-.921 1.902 0l1.519 4.674a1 1 0 0 0 .95.69h4.915c.969 0 1.371 1.24.588 1.81l-3.976 2.888a1 1 0 0 0-.363 1.118l1.518 4.674c.3.922-.755 1.688-1.538 1.118l-3.976-2.888a1 1 0 0 0-1.176 0l-3.976 2.888c-.783.57-1.838-.197-1.538-1.118l1.518-4.674a1 1 0 0 0-.363-1.118l-3.976-2.888c-.784-.57-.38-1.81.588-1.81h4.914a1 1 0 0 0 .951-.69l1.519-4.674z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <img
            class="ml-4"
            height="25"
            src="https://assets.coingecko.com/coins/images/6799/thumb/BSV.png"
            width="25"
          />
          <h3
            class="text-sm break-before-auto overflow-hidden ml-4"
          >
            Bitcoin SV
          </h3>
        </div>
      </li>
    </ul>
  </form>,
]
`;
